code/0010-define-calculations-of-existential-deposit.md

```diff
@@ -0,0 +1,149 @@
+# Calculation of all On-Chain Existential Deposit Values
+
+## Abstract
+
+This document aims to provide standards for the initial definition of all 
+on-chain Existential Deposit (ED) values. While defining these values, we also
+need to ensure that undefined behavior does not emerge from how these values 
+interact.
+
+## Background
+
+While the values defined for ED have mostly been consistent, we have so far 
+lacked an official standard for how these values are defined. We have also not
+ensured that ED values are accessible wherever needed in our runtime, node, and
+front-end applications. We must decide on a consistent standard to use for the
+initial definition of all ED values on our chain while also allowing for 
+extensibility in the future.
+
+## Requirements
+
+* ED values for all tokens supported by the chain are available AND consistent
+  across our runtime, node, and front-end applications.
+
+* When the ED of token A is swapped into token B, the new balance of token B is
+  substantial for ED. In other words, a user cannot "delete" their balance when 
+  they convert it into another token.
+
+* The implementation of our ED value retrieval should allow for the overwriting
+  of the standard ED value defined in this document.
+
+## Method
+
+### Initial Standard ED Value
+
+With the following definitions: 
+
+* `local_ed` - The functional ED of a asset on our chain.
+
+* `native_token_ed` - The ED of our chains native asset.
+
+* `foreign_asset_ratio` - The ratio of 'one in the foreign asset' over 'one in 
```

Comment:
that is not exchange. nobody exchanges nothing here. it may be even illegal for some users. it just pays in relevant amount. there is no exchange.

Reply:
Just because someone does not exchange it does not mean it's not a exchange rate. Eg: Your bank may show your account balance in USD even though you are actually saving in LKR.